models/user_reports.go

```diff
@@ -0,0 +1,5 @@
+package models
+
+type UserReport struct {
+	Reason string `json:"0"`
```

Comment:
interesting, are user reports objects with `0` as their key or are they an array of one element?

Reply:
hmm, that looks like an array to me. So just as a recap. This is how we unmarshall `user_reports`
```go
UserReports                []UserReport       `json:"user_reports"`
```
So `UserReports` is a slice of `UserReport`, where  each `UserReport` is a struct
```go
type UserReport struct {
 	Reason string `json:"0"`
}
```
Then shouldn't each `Reason` be an array of 4 elements (maybe it's an object if we look at the Reddit API docs?) with the example you gave above?
```json
[
      "testing tatata (mods please ignore this)",
      1,
      false,
      true
]
```
This denotes an array or an object? with types of `string, int, bool, bool`

Reply:
It does denote an array. I am unable to find out what the other variables refer to. Is there any reddit API documentation other than [this](https://www.reddit.com/dev/api)?

Reply:
@devmedoo Weird. The link you sent is the reddit doc that I based mira on. I believe it should be working then. Consider this resolved. Thanks

Reply:
I am unable to find any information regarding outputs of API calls on that page. It lists endpoints and their brief description and inputs. Am I missing something?

Reply:
Have you tried manually making a call with `MiraRequest` and seeing what Reddit returns for that endpoint and element?

Reply:
Did some digging. [Found](https://www.reddit.com/r/redditdev/comments/olqo5s/comment/h5g71gf/?utm_source=reddit&utm_medium=web2x&context=3) a reddit admin explaining each array element. 

I don't think my approach of using array index will work with the default json library as it is an array that needs to be unmarshalled into a struct. 
Maybe I should change the UserReports field type to `[]interface{}` then add a method to return user reports as an array of structs using casting?

Reply:
Nice find! Why can't we declare it as an array of structs and unmarshall it there? Even if most of the time it will be an array of size 1. Sounds like a good idea

Reply:
The ` [ ] ` in [the reddit API response](https://github.com/thecsw/mira/pull/12#discussion_r801937398) denotes its type as array. As far as I know `json.Unmarshal` default behavior does not convert JSON arrays to structs. [This is a similar example with a much better explanation.](https://stackoverflow.com/a/42378906)

Reply:
@devmedoo run this code, where we unmarshall a JSON array into structs

I made a playground example -> https://go.dev/play/p/4c2OSUAOTKA

```go
package main

import (
	"encoding/json"
	"fmt"
)

type MyStruct struct {
	Value1 string `json:"value_one"`
	Value2 int    `json:"value_two"`
	Value3 bool   `json:"value_three"`
}

func main() {
	vals := make([]MyStruct, 0, 10)

	json.Unmarshal([]byte(`[
{
  "value_one": "hello",
  "value_two": 37,
  "value_three": true
},
{
  "value_one": "this works?",
  "value_two": 9999,
  "value_three": false
}
]`), &vals)

	fmt.Println(vals)
}
```

Reply:
```
"user_reports": [
    [

    ]
]
```

It is an array of arrays. Your example is an array of objects. In JSON, square brackets `[]` denote arrays and curly brackets `{}` denote objects. Objects have a defined index (i.e: `"value_one"`). While arrays do not have a defined index and usually is iterated through numerically in programs. Converting your example JSON string to one that is an array of arrays instead of an array of objects in [this playground](https://go.dev/play/p/j5GPHhInzz-) does not provide the desired output. What I suggested in my previous comment works like [this](https://go.dev/play/p/X2hr3ZZRXRO).

Reply:
That makes sense. Thanks for clarifying that up! I think your approach here is the simplest one, therefore most optimal, and nice to read. Approved

Reply:
On second thought, since I can't change the data type for what should be
```go
UserReports                []interface{}       `json:"user_reports" 
````
into an array of UserReport during runtime, maybe I should just expose the report reason, count, snooze status and can snooze through methods of ReportListingChild?